funnel/views/section.py

```diff
@@ -59,39 +47,49 @@ def section_new(profile, project):
-    if form.validate_on_submit():
-        form.populate_obj(section)
-        db.session.commit()
-        flash(_("Your section has been edited"), 'info')
-        return redirect(project.url_for(), code=303)
-    return render_form(form=form, title=_("Edit section"), submit=_("Save changes"))
+@route('/<profile>/<project>/sections/<section>')
+class SectionView(UrlForView, ModelView):
+    model = Section
+    route_model_map = {'profile': 'project.profile.name', 'project': 'project.name', 'section': 'name'}
 
+    def loader(self, kwargs):
+        return self.model.query.join(Project).join(Profile).filter(
+                Project.name == kwargs.get('project'), Profile.name == kwargs.get('profile'),
+                Section.name == kwargs.get('section')
+            ).first_or_404()
 
-@app.route('/<profile>/<project>/sections/<section>/delete', methods=['GET', 'POST'])
-@funnelapp.route('/<project>/sections/<section>/delete', methods=['GET', 'POST'], subdomain='<profile>')
-@lastuser.requires_login
-@load_models(
-    (Profile, {'name': 'profile'}, 'g.profile'),
-    ((Project, ProjectRedirect), {'name': 'project', 'profile': 'profile'}, 'project'),
-    (Section, {'name': 'section', 'project': 'project'}, 'section'),
-    permission='delete-section')
-def section_delete(profile, project, section):
-    return render_delete_sqla(section, db, title=_(u"Confirm delete"),
-        message=_(u"Do you really wish to delete section ‘{title}’?").format(title=section.title),
-        success=_("Your section has been deleted"),
-        next=project.url_for('sections'),
-        cancel_url=project.url_for('sections'))
+    @route('', methods=['GET'])
+    @render_with('section.html.jinja2', json=True)
+    def view(self, **kwargs):
+        return {'project': self.obj.project, 'section': self.obj.current_access()}
+
+    @route('edit', methods=['GET', 'POST'])
+    def edit(self, **kwargs):
+        form = SectionForm(obj=self.obj, model=Section, parent=self.obj.parent)
+        if form.validate_on_submit():
+            form.populate_obj(self.obj)
+            db.session.commit()
+            flash(_("Your section has been edited"), 'info')
+            return redirect(self.obj.project.url_for('sections'), code=303)
+        return render_form(form=form, title=_("Edit section"), submit=_("Save changes"))
+
+    @route('delete', methods=['GET', 'POST'])
+    def delete(self, **kwargs):
+        return render_delete_sqla(self.obj, db, title=_(u"Confirm delete"),
+            message=_(u"Do you really wish to delete section '{title}’?").format(title=self.obj.title),
+            success=_("Your section has been deleted"),
+            next=self.obj.project.url_for('sections'),
+            cancel_url=self.obj.project.url_for('sections'))
+
+
+@route('/<project>/sections/<section>', subdomain='<profile>')
+class FunnelSectionView(SectionView):
+    pass
+
+
+SectionView.init_app(app)
+FunnelSectionView.init_app(funnelapp)
```

Comment:
Have you confirmed this works as expected? With different sets of URLs for each? Asking because I haven't.

Reply:
Yeah. Tested on both funnel and hasgeekapp. works.